server/image/image.go

```diff
@@ -0,0 +1,78 @@
+package image
+
+import (
+	"context"
+	imagepkg "github.com/argoproj/argo-cd/v2/pkg/apiclient/image"
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/authn/k8schain"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	log "github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type server struct {
+	keychain authn.Keychain
+}
+
+func (s *server) GetImage(ctx context.Context, req *imagepkg.GetImageRequest) (*imagepkg.GetImageResponse, error) {
+	log.WithField("image", req.Image).Info("GetImage")
+	ref, err := name.ParseReference(req.Image)
+	if err != nil {
+		return nil, err
+	}
+	img, err := remote.Image(ref, remote.WithAuthFromKeychain(s.keychain))
```

Comment:
Would we want to be able to support more than one container registry, e.g. a public one and then one or more private ones (e.g. a test and a prod internal registry)?

Reply:
How do we connect to the private repository? How is the keychain populated?